models/repo_list_test.go

```diff
@@ -60,10 +57,109 @@ func TestSearchRepositoryByName(t *testing.T) {
 		Page:     1,
 		PageSize: 10,
 		Private:  true,
-		Searcher: &User{ID: 14},
 	})
 
 	assert.NotNil(t, repos)
 	assert.NoError(t, err)
 	assert.Equal(t, int64(3), count)
+
+	// Get all public repositories by name
+	repos, count, err = SearchRepositoryByName(&SearchRepoOptions{
+		Keyword:  "big_test_",
+		Page:     1,
+		PageSize: 10,
+	})
+
+	assert.NotNil(t, repos)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(4), count)
+
+	// Get all public + private repositories by name
+	repos, count, err = SearchRepositoryByName(&SearchRepoOptions{
+		Keyword:  "big_test_",
+		Page:     1,
+		PageSize: 10,
+		Private:  true,
+	})
+
+	assert.NotNil(t, repos)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(8), count)
+
+	// Get all public repositories of user
+	repos, count, err = SearchRepositoryByName(&SearchRepoOptions{
+		Page:     1,
+		PageSize: 10,
+		OwnerID:  15,
+		Searcher: &User{ID: 15},
+	})
+
+	assert.NotNil(t, repos)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(3), count)
```

Comment:
@lunny @lafriks I think, this should return 2, not 3 (only directly owned repositories). Organization repositories should be included with `collaborate` flag. What do you think? Same applies to line 112. Or add somethink like `organization` flag.

Reply:
Orgnization repositories should be in collaborate, I think it's right.